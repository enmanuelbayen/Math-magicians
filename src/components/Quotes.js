import React, { useState, useEffect } from 'react';
import '../style/Calculator.css';

const Quotes = () => {
  const [quotes, setQuotes] = useState({});
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);

  useEffect(() => {
    const fetchQuote = async () => {
      setLoading(true);
      try {
        const response = await fetch(
          `https://api.api-ninjas.com/v1/quotes?category=${'learning'}`,
          {
            headers: {
              'X-Api-Key': '4cIYolx2izdKXSCL2c8Ow36G7YHL1z8gflMZ7nkt',
            },
          },
        );
        const json = await response.json();
        setQuotes(json[0]);
      } catch (error) {
        setError(true);
      }
      setLoading(false);
    };
    fetchQuote();
  }, []);

  if (error) {
    return (
      <p className="quoteError">Ups... Quote is missing!</p>
    );
  }
  if (loading) {
    return (
      <p className="quoteLoading">Loading...</p>
    );
  }

  return (
    <div className="quoteBox">
      {quotes && (
      <p className="dailyQuote">
        &quot;
        {quotes.quote}
        {' '}
        &quot; -
        <span>
          {' '}
          {quotes.author}
        </span>
      </p>
      )}
    </div>
  );
};

export default Quotes;
