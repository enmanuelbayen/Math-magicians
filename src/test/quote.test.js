import React from 'react';
import { render } from '@testing-library/react';
import Quotes from '../components/Quotes';

describe('Quotes component', () => {
  test('Renders correctly', () => {
    const { container } = render(<Quotes />);
    expect(container).toMatchSnapshot();
  });

  test('Displays loading message while fetching quotes', () => {
    const { getByText } = render(<Quotes />);
    const loadingMessage = getByText('Loading...');
    expect(loadingMessage).toBeInTheDocument();
    expect(loadingMessage).toMatchSnapshot();
  });

  test('Displays error message when quote fetch fails', () => {
    jest.spyOn(window, 'fetch').mockRejectedValueOnce(new Error('Fetch error'));
  
    const { getByText, queryByText } = render(<Quotes />);
    const loadingMessage = getByText('Loading...');
  
    expect(loadingMessage).toBeInTheDocument();
  
    setTimeout(() => {
      const errorMessage = queryByText('Ups... Quote is missing!');
  
      expect(loadingMessage).not.toBeInTheDocument();
      expect(errorMessage).toBeInTheDocument();
      expect(errorMessage).toMatchSnapshot();
    }, 0);
  });
});
