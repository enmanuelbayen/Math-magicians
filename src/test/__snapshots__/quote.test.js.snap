// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component Displays loading message while fetching quotes 1`] = `
<p
  class="quoteLoading"
>
  Loading...
</p>
`;

exports[`Quotes component Renders correctly 1`] = `
<div>
  <p
    class="quoteLoading"
  >
    Loading...
  </p>
</div>
`;
